torch/_dynamo/eval_frame.py

```diff
@@ -98,18 +98,35 @@ class Unset(Enum):
 unset = Unset.token
 
 
-def _maybe_set_eval_frame(callback: DynamoCallback):
+def _maybe_set_eval_frame(
+    callback: DynamoCallback, context: _TorchDynamoContext, state: str
+):
```

Comment:
I think we can achieve this without modifying this function. Instead of passing in context and state into here, and overriding the `callback` used in `set_eval_frame(callback)`.

We can pass in the overriden callback from the callsites e.g. `_maybe_set_eval_frame(context.maybe_warmup(callback))`